modules/infra/main.tf

```diff
@@ -227,18 +243,25 @@ resource "azurerm_network_security_group" "bosh_deployed_vms_security_group" {
 # ============= Networking
 
 resource "azurerm_virtual_network" "pcf_virtual_network" {
-  name                = "${var.env_name}-virtual-network"
-  depends_on          = ["azurerm_resource_group.pcf_resource_group"]
-  resource_group_name = "${azurerm_resource_group.pcf_resource_group.name}"
+  count               = "${var.create_vnet ? "1" : "0"}"
+  name                = "${var.vnet_name != "" ? "${var.vnet_name}" : "${var.env_name}-virtual-network"}"
+  depends_on          = ["azurerm_resource_group.pcf_network_rg"]
+  resource_group_name = "${azurerm_resource_group.pcf_network_rg.name}"
   address_space       = "${var.pcf_virtual_network_address_space}"
   location            = "${var.location}"
 }
 
+data "azurerm_virtual_network" "pcf_virtual_network" {
```

Comment:
@jseely I'm not used to seeing `data` and `resource` blocks defined with the same type and name without having inverse `count`s (i.e. having one or the other but not both). I'm actually a little surprised that this doesn't cause an error... Did you have a particular reason for doing it this way? Or is this a pattern you've seen before?

Reply:
This is intentional. The `data` block depends on the `resource` block to ensure the resource was either previously created or was created by the `resource` block. This allows all the rest of the resources to reference the `data` block. This would error out if the `data` block didn't depend on the `resource` block.